examples/download_everything.py

```diff
@@ -0,0 +1,272 @@
+            )
+            os.makedirs(result_path)
+
+            result.params.run_id = run.params.run_id
+
+            result.read()  # get full result
+
+            with open(os.path.join(result_path, "result.json"), "w") as f:
+                json.dump(result.params.to_dict_full(), f, indent=4)
+
+            logs_path = os.path.join(result_path, "logs")
+            os.makedirs(logs_path)
+
+            logs = {
+                "webrtc.txt": result.params.log_paths.webrtc,
+                "selenium.txt": result.params.log_paths.selenium,
+                "browser.txt": result.params.log_paths.browser,
+                "rru.json": result.params.log_paths.rru,
+                "allure.tar.gz": result.params.log_paths.allure_report,
+            }
+
+            for log_name, log_url in logs.items():
+                if not log_url:
+                    continue
+
+                r = requests.get(
+                    log_url,
+                    headers={"Authorization": f"LoaderoAuth {access_token}"},
+                    stream=True,
+                )
+
+                with open(os.path.join(logs_path, log_name), "wb") as fd:
+                    for chunk in r.iter_content(chunk_size=128):
+                        fd.write(chunk)
+
+            artifacts = {
+                "audio": result.params.artifacts.audio.paths,
+                "downloads": result.params.artifacts.downloads.paths,
+                "screenshots": result.params.artifacts.screenshots.paths,
+                "video": result.params.artifacts.video.paths,
+            }
+
+            artifacts_path = os.path.join(result_path, "artifacts")
+            os.makedirs(artifacts_path)
+
+            for artifact_type, artifact_urls in artifacts.items():
+                if not artifact_urls:
+                    continue
+
+                if len(artifact_urls) == 0:
+                    continue
+
+                artifact_type_path = os.path.join(artifacts_path, artifact_type)
+                os.makedirs(artifact_type_path)
+
+                for url in artifact_urls:
+                    r = requests.get(
+                        url,
+                        headers={
+                            "Authorization": f"LoaderoAuth {access_token}"
+                        },
+                        stream=True,
+                    )
+
+                    with open(
+                        os.path.join(artifact_type_path, os.path.basename(url)),
+                        "wb",
+                    ) as fd:
+                        for chunk in r.iter_content(chunk_size=128):
+                            fd.write(chunk)
+
+
+def get_tests(output_dir: str, limit: int):
+    tests_path = os.path.join(output_dir, "tests")
+    os.makedirs(tests_path)
+
+    project = Project().read()
+    language = project.params.language
+
+    for test in project.tests(
+        query_params=QueryParams().limit(limit).filter("order_by", "-id")
+    )[0]:
```

Comment:
imo for runs the syntax was better, otherwise it looks like only the first test is considered

Reply:
fix the same for asserts and preconditions